### PR TITLE
fix(Pagination): remove spin buttons

### LINKED ISSUE #9218: Bug - Pagination - spin buttons on number input
Pagination: spin buttons should not be displayed (they are not shown in core), it is causing spacing issues. Also the input field seems small.

Screenshot 2023-06-01 at 11 19 28 (2)

**Expected behavior**
It should look like in the core implementation:
Screenshot 2023-06-01 at 11 48 26 (2)

### PR DESCRIPTION
**Pull request by adamviktora**
**What**: Closes #9218, #9187

### CODE REVIEW

**Review by gitdallas — APPROVED** [review]

**Review by mcarrano — APPROVED** [review]
Looks good. Thanks @adamviktora !

**Review by tlabaj — APPROVED** [review]

**Review by mcoker — APPROVED** [review]
This fixes the base styling, but it's worth noting that if this does not use `<TextInput>`, all changes to the core form control component will need to be maintained here separately.

For example, another thing that isn't working properly in this PR is the disabled styling, because the visual presentation for a disabled form control changed with v5 (the `styles.formControl` div also needs the class `pf-m-disabled`). You can see in v4 that the [no items and one page example example inputs are disabled](https://www.patternfly.org/v4/components/pagination#no-items) and they are [not disabled in this PR build](https://patternfly-react-pr-9219.surge.sh/components/pagination#no-items).

If we can't use `<TextInput>` here for some reason, it would be good to review this a little more to make sure there is full parity with the form control component.

File: packages/react-core/src/components/Pagination/Navigation.tsx [hunk 1]
```diff
@@ -217,11 +218,10 @@ export class Navigation extends React.Component<NavigationProps, NavigationState
         </div>
         {!isCompact && (
           <div className={styles.paginationNavPageSelect}>
-            <input
-              className={css(styles.formControl)}
+            <TextInput
```
Comment (mcoker) [inline packages/react-core/src/components/Pagination/Navigation.tsx:221]: 🥳

**Review by mcoker — APPROVED** [review]
L🎃TM!

**Pull request merged**